remoting/subsystem/src/main/java/org/jboss/as/remoting/RemotingSubsystemXMLPersister.java

```diff
@@ -133,7 +148,19 @@ private void writeWorkerThreadPoolIfAttributesSet(final XMLExtendedStreamWriter
     }
 
     private void writeEndpointIfAttributesSet(final XMLExtendedStreamWriter writer, final ModelNode model) throws XMLStreamException {
-        RemotingSubsystem30Parser.ENDPOINT_PARSER.persist(writer, model);
+        boolean defined = false;
+        for (String adName : RemotingEndpointResource.ATTRIBUTES.keySet()) {
```

Comment:
Why complicate this much?

Given this is new version of model, lets just introduce new version of XSD as well.
and have clean parser / persister for current version of the subsystem.

Reply:
Because I don't want to do that now. It means a complete conversion and I don't want us to spend our time on that right now.

Reply:
Also I don't think the xsd is wrong. The endpoint element is a valid wrapper for a set of elements that are a natural attribute group. I wouldn't want to stick all the attributes on the root xml element. 

The thing I don't want to spend time on now is converting the whole subsystem to PersistentResourceXmlDescription.